rc/windowing/iterm.kak

```diff
@@ -18,7 +18,9 @@ define-command -hidden -params 2.. iterm-terminal-split-impl %{
                 if [ "$i" = '' ]; then
                     printf "'' "
                 else
-                    printf %s "$i" | sed -e "s|'|'\\\\''|g; s|^|'|; s|$|' |"
+                    printf "'"
+                    printf %s "$i" | sed -e "s|'|'\\\\''|g"
+                    printf "' "
```

Comment:
Sure, I could fix that as well. Do you think doing it like this or with sed is better?

Also, I noticed that only iterm-terminal-split-impl escapes the PATH and TMPDIR variables, is there a reason the other commands below it don't?

Reply:
> Also, I noticed that only iterm-terminal-split-impl escapes the PATH and TMPDIR variables

nice catch

> Do you think doing it like this or with sed is better?

I usually use

	printf "'%s' " "$(printf %s "$i" | sed "s|'|'\\\\''|g")"

I'm not sure which is better, depends on reader expectations.

I also noticed that, at least with the fixed logic you can remove the `if [ "$i" = '' ]; then` special case since it's subsumed by the general case.

Reply:
> > Do you think doing it like this or with sed is better?
> 
> I usually use
> 
> ```
> printf "'%s' " "$(printf %s "$i" | sed "s|'|'\\\\''|g")"
> ```

Oh yeah of course, that's a lot better than the three commands I have right now.

> I also noticed that, at least with the fixed logic you can remove the `if [ "$i" = '' ]; then` special case since it's subsumed by the general case.

Ah, nice. I thought so but wasn't entirely sure.